BEGINNERS.md

```diff
@@ -0,0 +1,309 @@
+# Minimal Snowdrift set up instructions
+
+This is a small tutorial on contributing to Snowdrift. You don't have to
+be tech savvy, but it helps. This is a minimal-ish description of what
+you need to do to get Snowdrift running locally and to start making and
+submitting updates.
+
+Although we recommend deeper study to understand how all of this works,
+you *can* simply follow the steps below, and it will work well enough
+to get started.
+
+Advanced users should adapt any of these instructions as seen fit.
+
+These instructions assume you use Debian testing or some variant of Ubuntu.
+Our general [GUIDE](GUIDE.md) has some notes for other systems.
+
+If you need *any* help at any point, come say "hi" at our freenode.net IRC
+channel [#snowdrift](https://snowdrift.coop/p/snowdrift/w/irc).
+We are always happy to assist and answer *any* questions!
+
+## Installing
+
+Open up a terminal, and run these commands, exactly as you see them. You
+could even copy and paste all of them into your terminal at once, and it
+would work.
+
+    sudo -i
+    aptitude update
+    aptitude install curl git postgresql postgresql-client
+    cd /usr/lib/x86_64-linux-gnu
+    ln -s libgmp.so.10 libgmp.so
```

Comment:
@wolftune There are two ways to compile a program that uses a library

* Include the entire library in the binary file. This is called "static linking". Most proprietary programs are statically linked.
* Put the library's shared object code into a `.so` file, so that multiple programs can just point to it. 

Dynamic linking helps with portability, and cuts down on hard disk usage.

libgmp is the GNU Multiple Precision library. It's needed to do decimal calculations with a high degree of precision. `libgmp.so.10` is the default location of the libgmp shared object file. However, many programs, not just `ghc`, dynamically link to `/usr/lib/x86_64-linux-gnu/libgmp.so`, not *necessarily* `/usr/lib/x86_64-linux-gnu/libgmp.so.10`. The GHC instructions hinted that this line might be needed, but didn't explicitly say it, so I included it anyway. It doesn't hurt.